### PR TITLE
[nlohmann-json] Update to 3.11.2

### LINKED ISSUE #26103: [nlohmann-json] update to 3.11.0
Library name: nlohmann-json

New version number: 3.11.0

Other information that may be useful (release notes, etc...)
https://github.com/nlohmann/json/releases/tag/v3.11.0

### PR DESCRIPTION
**Pull request by falbrechtskirchinger**
- #### What does your PR fix?
  Adds the latest version of `nlohmann-json` https://github.com/nlohmann/json/releases/tag/v3.11.2.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Unchanged. No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by falbrechtskirchinger** [comment]
A number of issues have been found in v3.11.1 (including those in this CI run) and v3.11.2 should be released soon. I'll retry then.

**Comment by falbrechtskirchinger** [comment]
This last force-push is obviously only for testing purposes. I want to make sure that v3.11.2 + the pdal port changes do in fact fix the previous CI issues.

**Comment by falbrechtskirchinger** [comment]
Updated to 3.11.2. The issues compiling `json-schema-validator` and `pdal` have been addressed.

File: ports/nlohmann-json/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "nlohmann-json",
-  "version-semver": "3.10.5",
-  "port-version": 3,
+  "version-semver": "3.11.2",
```
Comment (JonLiu1993) [inline ports/nlohmann-json/vcpkg.json:3]: `These macros are defined by the library and contain the version numbers according to [Semantic Versioning 2.0.0](https://semver.org/spec/v2.0.0.html).`
https://github.com/nlohmann/json/blob/6a7392058e69956d610acefa7e3ce1cd60c8f5a6/docs/mkdocs/docs/api/macros/nlohmann_json_version_major.md#nlohmann_json_version_major-nlohmann_json_version_minor-nlohmann_json_version_patch

Comment (falbrechtskirchinger) [reply ports/nlohmann-json/vcpkg.json:3]: What's the issue here?

Comment (JonLiu1993) [reply ports/nlohmann-json/vcpkg.json:3]: There is no problem, I just listed the reasonable use of "version-semver" to help my colleagues make sure that it is used correctly when merging

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**